Assets/scripts/NoteObject.cs

```diff
@@ -3,26 +3,26 @@
 
 public class NoteObject : MonoBehaviour {
 
-	public Note NoteData;
-	// TODO: Use object pool for better efficiency
+	public Note NoteData { get { return _NoteData; } }
+	private Note _NoteData;
 	public float destroyY = -10f; // Y coordinate under which object will self destruct
 	public Vector3 velocity = new Vector3(0, -0.22f, 0);
 
 	private void Start() {
-		if (NoteData != null) {
+		if (_NoteData != null) {

```

Comment:
This probably does nothing because _Notedata can't be set before start gets called. My b for not deleting it earlier